### DISCUSSION TITLE
[General] Account Abstraction Transaction Verification Restriction

### DISCUSSION THREAD

**Discussion by estarriolvetch** [zkSync-Community-Hub/zksync-developers#130] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I am writing this to discuss the limitations in the verification step, specifically the restriction on touching only the storage slot of the account itself, which prevents many applications, including ours, from using this feature.  

Ref: https://era.zksync.io/docs/reference/concepts/account-abstraction.html#limitations-of-the-verification-step

These limitations include:
- Using the beacon proxy for this purpose. The beacon proxy needs to access the storage slot in the beacon to function correctly.
- Utilizing information from a register while validating the signature. For instance, checking if a hash is also co-signed by a centralized signer for 2FA.

### What is the specific issue or error you're encountering?

N/A

### Can you share the error messages or logs you're receiving, if any?

N/A

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

N/A

**Comment by idea404** [comment, marked as answer]
Hello @estarriolvetch,

Thanks for highlighting your concerns about zkSync's account abstraction verification limitations.

I understand the current restrictions, especially on accessing only an account's storage slots during verification, impact certain applications like yours. While these limitations are in place for security reasons, the team is actively considering extending the accessible slots to include aspects like ERC20 balances.

We're continuously refining zkSync based on feedback and real-world use cases. For now, I'd recommend exploring alternative design patterns within the current constraints. Your feedback is highly valuable, and we'll certainly consider it as we plan future updates.

**Reply by idea404** [reply]
Please also feel free to open a feature request on https://github.com/matter-labs/zksync-era

**Comment by amyliasnow** [comment]
Thank you for bringing up this issue regarding the limitations in the verification step of account abstraction. It's important to consider these limitations when designing and implementing applications on zkSync. If you have specific use cases or suggestions for addressing these limitations, feel free to share them so that the community can discuss potential solutions or workarounds.

**Comment by ryangrullon** [comment]
The limitations in the verification step, particularly regarding the restriction on touching only the storage slot of the account itself, pose challenges for applications requiring more complex interactions, such as utilizing beacon proxies or incorporating additional validation logic during signature checks.

**Comment by reginaldolivas** [comment]
The limitations in the verification step of account abstraction can pose challenges for certain applications, particularly those requiring access to additional storage slots or external information during signature validation.

**Comment by postm4ltolin** [comment]
The limitations on the verification step in zkSync, specifically the restriction on touching only the storage slot of the account itself, can make it difficult for some applications to use this feature. This can include using the beacon proxy or utilizing information from a register while validating the signature. While these limitations may pose challenges for certain use cases, they are in place to ensure the security and integrity of the zkSync network.

**Comment by moyemv** [comment]
The user is discussing the limitations of the verification step in the Semaphore protocol on zkSync, specifically the restriction on touching only the storage slot of the account itself. The user has not reported encountering any specific issue or error and has not provided any relevant code or repository link. Further discussion may be needed to address the limitations of the verification step and potential solutions.
